### ISSUE TITLE
Standardize inputs for `syntheval` metrics

### ISSUE THREAD

**Issue by jhseeman** [UrbanInstitute/syntheval#89]
Currently, `syntheval` functions accept `postsynth` arguments as arbitrary dataframes or `postsynth` objects from `tidysynthesis`. As we extend `syntheval` for more features, like handling multiple replicates (for example, #86 #80), we'd like to standardize inputs to all `syntheval` functions. This issue proposes the `eval_data` object to do that:

```
eval_data(
    conf_data, # confidential dataframe
    synth_data, # either 1 dataframe, 1 postsynth, or a list of multiples of either
    holdout_data = NULL,  # optional holdout data
)
```

**Comment by awunderground** [comment]
I think there are two downsides to consider:

1. This introduces an extra step for users before they see results. But there is upside!
2. This makes it really difficult to manipulate (`filter()`, `select()`, `mutate()`, etc.) the data before calculating a specific metric.
    - Suppose I want to look at `util_moments()` after removing one quirky record from the `conf_data`. 
    - Suppose I want to sum three variables in the `conf_data` and `synth_data` before using discriminant metrics. 
    - Suppose I want to pick a subset of variables for discriminant metrics or disclosure risk metrics using `select()`. 

Can we make it so the metrics take `post_synth`/`conf_data` OR and `eval object? Do you have other ideas for getting around #2?

**Comment by jhseeman** [comment]
>  This makes it really difficult to manipulate (`filter()`, `select()`, `mutate()`, etc.) the data before calculating a specific metric.

Not sure I agree about the difficulty- in most cases, this is one line of boilerplate code that would follow any data manipulations and precede any arbitrary set of `syntheval` metrics the user wants to run. For descriptive metrics, it doesn't seem like a good practice to encourage using modified copies of `eval_data` for different specific metrics. For model-based metrics like discriminant-based ones, the transformations you suggested could be handled by `recipes`, yes? Or am I mistaken? 

> Can we make it so the metrics take `post_synth`/`conf_data` OR and `eval object? Do you have other ideas for getting around 2.?
The point of this object is to encapsulate input processing logic to avoid boilerplate code in each metric, but this solution adds more, not less!

**Comment by awunderground** [comment]
Closed with #90
